src/Fl_Message.cxx

```diff
@@ -327,6 +340,25 @@ int Fl_Message::innards(const char *fmt, va_list ap, const char *b0, const char
   else
     message_->labelsize(fl_message_size_);
 
+  window_->color(message_color_);
+  message_color(message_color_default_); // reset
+  message_->labelcolor(message_labelcolor_);
+  message_labelcolor(message_labelcolor_default_); // reset
+
+  for (int b = 2; b >= 0; b--) {
+    button_[b]->color(message_button_color_);
+    message_button_color(message_button_color_default_); // reset
```

Comment:
This seems to reset the button color to the default value inside the loop which implies that only one button color would ever be changed. This and all the following "reset" statements should IMHO be outside (below/after) the loop.
Note: this is not a full review, it's just something I stumbled across.